app/models/product.rb

```diff
@@ -24,28 +63,17 @@ def average_rating
     return average_rating
   end
 
-  def has_available_stock?
-    (quantity_tied_up_in_pending_transactions + 1) <= stock
-  end
-
-  def self.top_products
-    sorted_products = Product.all.sort_by { |product| product.average_rating }.reverse!
-    top_products = sorted_products[0..11]
-  end
-
-  def update_stock
-    # code to reduce stock when order is paid or shipped
-    # note: if we go with shipped / complete, the private method
-    # quantity_tied_up_in_pending_transactions will need to be updated to
-    # account for paid orders as well -J
+  def stock?
+    stock > 0
```

Comment:
After seeing how many things renaming #stock? touches, feel free to leave this! lols.